alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -0,0 +1,529 @@
+	}
+
+	if relatedClusterId, ok := d.GetOk("related_cluster_id"); ok {
+		request.RelatedClusterId = relatedClusterId.(string)
+	}
+
+	if optionSoftware, ok := d.GetOk("option_software_list"); ok {
+		var softwareList []string
+		for _, vv := range optionSoftware.([]interface{}) {
+			softwareList = append(softwareList, vv.(string))
+		}
+		request.OptionSoftWareList = &softwareList
+	}
+
+	vpcService := VpcService{client}
+	vswitchId := Trim(d.Get("vswitch_id").(string))
+	request.NetType = "classic"
+
+	if vswitchId != "" {
+		request.VSwitchId = vswitchId
+		request.NetType = "vpc"
+
+		vsw, err := vpcService.DescribeVSwitch(vswitchId)
+		if err != nil {
+			return WrapError(err)
+		}
+
+		if request.ZoneId == "" {
+			request.ZoneId = vsw.ZoneId
+		} else if request.ZoneId != vsw.ZoneId {
+			return WrapError(Error("The specified vswitch %s isn't in the zone %s.", vsw.VSwitchId, request.ZoneId))
+		}
+
+		request.VpcId = vsw.VpcId
+	}
+
+	var hostGroups []emr.CreateClusterV2HostGroup
+	if groups, ok := d.GetOk("host_group"); ok {
+		for _, group := range groups.(*schema.Set).List() {
+			kv := group.(map[string]interface{})
+			hostGroup := emr.CreateClusterV2HostGroup{}
+
+			if v, ok := kv["period"]; ok {
+				hostGroup.Period = v.(string)
+			}
+
+			if v, ok := kv["sys_disk_capacity"]; ok {
+				hostGroup.SysDiskCapacity = v.(string)
+			}
+
+			if v, ok := kv["disk_capacity"]; ok {
+				hostGroup.DiskCapacity = v.(string)
+			}
+
+			if v, ok := kv["sys_disk_type"]; ok {
+				hostGroup.SysDiskType = v.(string)
+			}
+
+			if v, ok := kv["disk_type"]; ok {
+				hostGroup.DiskType = v.(string)
+			}
+
+			if v, ok := kv["host_group_name"]; ok {
+				hostGroup.HostGroupName = v.(string)
+			}
+
+			if v, ok := kv["disk_count"]; ok {
+				hostGroup.DiskCount = v.(string)
+			}
+
+			if v, ok := kv["auto_renew"]; ok {
+				if v.(bool) == false {
+					hostGroup.AutoRenew = "false"
+				} else if v.(bool) == true {
+					hostGroup.AutoRenew = "true"
+				}
+			}
+
+			if v, ok := kv["gpu_driver"]; ok {
+				hostGroup.GpuDriver = v.(string)
+			}
+
+			if v, ok := kv["node_count"]; ok {
+				hostGroup.NodeCount = v.(string)
+			}
+
+			if v, ok := kv["instance_type"]; ok {
+				hostGroup.InstanceType = v.(string)
+			}
+
+			if v, ok := kv["charge_type"]; ok {
+				hostGroup.ChargeType = v.(string)
+			}
+
+			if v, ok := kv["host_group_type"]; ok {
+				hostGroup.HostGroupType = v.(string)
+			}
+
+			hostGroups = append(hostGroups, hostGroup)
+		}
+	}
+	request.HostGroup = &hostGroups
+
+	raw, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+		return emrClient.CreateClusterV2(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_cluster", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	response, _ := raw.(*emr.CreateClusterV2Response)
+	d.SetId(response.ClusterId)
```

Comment:
After creating a cluster, the method should wait for cluster is Running before returning.

Reply:
fixed.